client/my-sites/domains/domain-search/index.jsx

```diff
@@ -142,15 +145,23 @@ class DomainSearch extends Component {
 		if ( this.props.domainAndPlanUpsellFlow ) {
 			// If we are in the domain + annual plan upsell flow, we need to redirect
 			// to the plans page next and let it know that we are still in that flow.
-			this.props.shoppingCartManager.addProductsToCart( [ registration ] ).then( () => {
-				page( `/plans/${ this.props.selectedSiteSlug }?domainAndPlanPackage=true` );
-			} );
+			this.props.shoppingCartManager
+				.addProductsToCart( [ registration ] )
+				.then( () => {
+					page( `/plans/${ this.props.selectedSiteSlug }?domainAndPlanPackage=true` );
+				} )
+				.catch( () => {
+					// Nothing needs to be done here. CartMessages will display the error to the user.
+				} );
 			return;
 		}
 
 		this.props.shoppingCartManager
 			.addProductsToCart( [ registration ] )
-			.then( () => page( domainAddEmailUpsell( this.props.selectedSiteSlug, domain ) ) );
+			.then( () => page( domainAddEmailUpsell( this.props.selectedSiteSlug, domain ) ) )
```

Comment:
any errors raised by `page` and `domainAddEmailUpsell` would be captured by the `catch()` below, is that intended? This is similarly performed in other files in this same PR.

Reply:
🤔 good point. I'll move the `catch()` up.

Reply:
Ah, crap. I just realized that this won't do what I want.

`Promise.reject().catch(() => {}).then(() => console.log('success'))` will print 'success' and I want the rejection to still skip the `then`.

Reply:
Ok, I think I came up with a better solution using async/await.